python/ql/src/experimental/Security/BufferedRWPairUndefinedBehavior.ql

```diff
@@ -0,0 +1,23 @@
+/**
+ * @name Same parameter used as reader and writer in BufferedRWPair
+ * @description Use of same parameter as reader and writer in BufferedRWPair.
+ * @kind problem
+ * @problem.severity warning
+ * @precision medium
+ * @id python/bufferedrwpair-undefined-behavior
+ * @tags reliability
+ *       security
+ */
+
+import python
+
+ClassValue requestFunction() { result = Module::named("io").attr("BufferedRWPair") }
+
+from CallNode call, ClassValue func, ControlFlowNode read, ControlFlowNode write
+where
+  func = requestFunction() and
+  func.getACall() = call and
+  read = call.getArg(0) and
+  write = call.getArg(1) and
+  read.getNode().toString() = write.getNode().toString()
```

Comment:
You should not be using `toString` in queries and libraries, as the internal string representation may change at any moment.
Also, it's fairly imprecise to look at the _names_ of these arguments, as they may be aliased:
```py
reader = io.BufferedReader(io.RawIOBase)
also_reader = reader
pair1 = io.BufferedRWPair(reader, also_reader)
```
A better solution would be to see if they point to the same value: `read.pointsTo() = write.pointsTo()`.

Reply:
Done, thanks for the tip!